homeassistant/components/transmission/sensor.py

```diff
@@ -81,8 +81,34 @@ def available(self):
     @property
     def device_state_attributes(self):
         """Return the state attributes, if any."""
-        if self._tm_client.api.started_torrent_dict and self.type == "started_torrents":
-            return {STATE_ATTR_TORRENT_INFO: self._tm_client.api.started_torrent_dict}
+        if self.type == "started_torrents":
+            return {
+                STATE_ATTR_TORRENT_INFO: _torrents_info(
```

Comment:
The top key is the same in all cases. Just create the nested dict infos first and then set it in the main dict for key `STATE_ATTR_TORRENT_INFO` once and return that.

Reply:
@MartinHjelmare Not at all, in case when the sensor has `type == 'current_status'` or `upload_speed`, this attribute should not be set. Current implementation of the sensor is a bit messy, so I'd prefer splitting it into 3 classes:
- TransmissionSpeedSenor - for the speed sensors, will not have `torrents_info`.
- TranmissionTorrentsSensor - for the torrents sensors, will have `torrents_info`.
- TransmissionStatus sensor - for the status sensor, will not have `torrents_info`

This will make it possible to drop `type`, simplify the implementation a lot and it will be much easier to work with the `torrents_info` attribute.

@JPHutchins @engrbm87 What do you think about such kind of refactoring?